packages/block-library/src/columns/edit.native.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+import { dropRight, times } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { PanelBody, StepperControl } from '@wordpress/components';
+import {
+	InspectorControls,
+	InnerBlocks,
+	BlockControls,
+	BlockVerticalAlignmentToolbar,
+} from '@wordpress/block-editor';
+import { withDispatch, useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { useResizeObserver } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+
+/**
+ * Allowed blocks constant is passed to InnerBlocks precisely as specified here.
+ * The contents of the array should never change.
+ * The array should contain the name of each block that is allowed.
+ * In columns block, the only block we allow is 'core/column'.
+ *
+ * @constant
+ * @type {string[]}
+ */
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+/**
+ * Number of columns to assume for template in case the user opts to skip
+ * template option selection.
+ *
+ * @type {number}
+ */
+const DEFAULT_COLUMNS = 2;
+const MIN_COLUMNS_NUMBER = 1;
+
+function ColumnsEditContainer( {
+	attributes,
+	updateAlignment,
+	updateColumns,
+	columnCount,
+	isSelected,
+	onAddNextColumn,
+	onDelete,
+} ) {
+	const [ resizeListener, sizes ] = useResizeObserver();
+	const [ columnsSettings, setColumnsSettings ] = useState( {
+		columnsInRow: MIN_COLUMNS_NUMBER,
+	} );
+
+	const { verticalAlignment } = attributes;
+	const { width } = sizes || {};
+
+	useEffect( () => {
+		const newColumnCount = ! columnCount ? DEFAULT_COLUMNS : columnCount;
+		updateColumns( columnCount, newColumnCount );
+		setColumnsSettings( {
+			...columnsSettings,
+			columnsInRow: getColumnsInRow( width, newColumnCount ),
+		} );
+	}, [ columnCount ] );
+
+	useEffect( () => {
+		setColumnsSettings( {
+			columnsInRow: getColumnsInRow( width, columnCount ),
+			width,
+		} );
+	}, [ width ] );
+
+	const getColumnsInRow = ( containerWidth, columnsNumber ) => {
+		if ( containerWidth < 480 ) {
+			return 1;
+		}
+		if ( containerWidth >= 480 && containerWidth < 768 ) {
+			return Math.min( columnCount, 2 );
+		}
+		return columnsNumber;
+	};
```

Comment:
I keep it because Lukes hook in initial version gives opportunity to replace it. It mean to be similar to `useViewportMatch` but instead of screen size it should take the container width read from onLayout. In final version this hook works quite different (it is mobile implementation of `useResizeObserver` hook) that we expect at the beginning so the values left as it was.

I'm not sure if we have any BREAKPOINTS object exposed that can be import here (correct me if I'm wrong). 

What I'm going to do is elevate 480 and 768 to const BREAKPOINTS objet at the top of Columns file like e.g in Media-Text. 

However I think in separate task we could try avoid declaring this BREAKPOINTS object because we already has the breakpoint defined e.g in `_breakpoints.scss` or in mentioned `useViewportMatch` hook above

Reply:
@dratwas @pinarol -  wdyt about proposed solution to replace hardcoded value ?